napytau/gui/gui_app.py

```diff
@@ -0,0 +1,478 @@
+        for i in range(len(self.datapoints)):
+            x, y = self.datapoints[i]
+
+            checkbox = customtkinter.CTkCheckBox(
+                self.frame_datapoint_checkboxes,
+                text=f"({x} | {y})",
+                variable=customtkinter.IntVar(value=1),
+                command=lambda index=i: self._data_checkbox_calculation_event(index))
+            checkbox.grid(row=i + 1, column=1, padx=35, pady=2, sticky="nsew")
+
+    def _data_checkbox_calculation_event(self, index: int) -> None:
+        """
+        Do not call from outside. Is called if a data checkbox for the calculation
+        of tau and delta-tau is called.
+        Toggles the intern boolean value of the datapoint.
+        :param index: Index of the pressed data checkbox.
+        """
+        self.datapoints_for_calculation[index].is_checked = \
+            not self.datapoints_for_calculation[index].is_checked
+        if self.datapoints_for_calculation[index].is_checked:
+            print("[calculation] checkbox with index " + str(index) + " activated.")
+        else:
+            print("[calculation] checkbox with index " + str(index) + " deactivated.")
+
+    def update_data_checkboxes(self, new_datapoints: List[Tuple[float, float]]) -> None:
+        """
+        Updates the datapoint for the gui and updates both columns of the
+        data checkboxes.
+        Call this method if there are new datapoints.
+        :param new_datapoints: The new list of datapoints.
+        """
+        self.datapoints = new_datapoints.copy()
+
+        for point in new_datapoints:
+            self.datapoints_for_fitting.append(CheckboxDataPoint(point, True))
+            self.datapoints_for_calculation.append(CheckboxDataPoint(point, True))
+
+        self._update_data_checkboxes_fitting()
+        self._update_data_checkboxes_calculation()
+
+    def change_scaling_event(self, new_scaling: str) -> None:
+        new_scaling_float = int(new_scaling.replace("%", "")) / 100
+        customtkinter.set_widget_scaling(new_scaling_float)
+
+
+    def calc(self) -> None:
+        entry_value = self.tau.get()
+
+        self.label.configure(text=f"Result: {logic(entry_value)}")
+
+
+def init(cli_arguments: CLIArguments) -> None:
+    app = GUIApp()
+    app.mainloop()
+
+def plot(value: int, window: GUIApp) -> Canvas:
+    """
+    Plot the graph.
+    :param value: The value.
+    :param window: The window.
+    :return: The canvas.
+    """
+    # the figure that will contain the plot
+    fig = Figure(figsize=(3, 2), dpi=100, facecolor="white", edgecolor="black")
+
+    # list of squares
+    y = [(i - 50) ** value for i in range(101)]
+
+    # adding the subplot
+    plot1 = fig.add_subplot(111)
+
+    # plotting the graph
+    plot1.plot(y)
+
+    # creating the Tkinter canvas
+    # containing the Matplotlib figure
+    canvas = FigureCanvasTkAgg(fig, master=window)
+    canvas.draw()
+
+    return canvas.get_tk_widget()
```

Comment:
I will leave these comments to you, they don't seem terribly trivial but we don't really need all of them, do as you see fit.

Reply:
Same as above, this is some leftover code from the graph mockup and will be replaced soon.